lib/SyTest/HTTPServer/Request.pm

```diff
@@ -48,6 +48,19 @@ sub respond_json
    $self->respond( $response );
 }
 
+sub respond_error_json
```

Comment:
Rather than add a new method, I'd suggest a named option on the existing `respond_json`. Change it to

    my ( $json, %opts ) = @_;

    my $response = HTTP::Response->new( $opts{code} // 200 );

Then elsewhere you can just

    ->respond_json( {}, code => 404 );